text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,454 @@
+to 63.
+
+If `sampled` is true with `p<63`, drop `p` from the outgoing context
+to convey unknown head probability.  Set `log_head_adjusted_count` to
+0.
+
+## Prototype
+
+[This proposal has been prototyped in the OTel-Go
+SDK.](https://github.com/open-telemetry/opentelemetry-go/pull/2177) No
+changes in the OTel-Go Tracing SDK's `Sampler` or `tracestate` APIs
+were needed.
+
+## Trade-offs and mitigations
+
+### Naming question
+
+This proposal changes the logic of the `TraceIDRatioBased` sampler,
+currently part of the OpenTelemetry specification, in a way that makes
+the name no longer meaningful.  The proposed sampler may be named
+`ConsistentSampler` and the existing `TraceIDRatioBased` sampler can
+be deprecated.
+
+Many SDKs already implement the `TraceIDRatioBased` sampler and it has
+been used for probability sampling at trace roots with arbitrary
+(i.e., not power-of-two) probabilities.  Because of this, we may keep
+the current (under-specified) `TraceIDRatioBased` sampler and rename
+it `ProbabilitySampler` to convey that it does behave in a specified
+way with respect to the bits of the TraceID.
+
+### Not using TraceID randomness
+
+It would be possible, if TraceID were specified to have at least 61
+uniform random bits, to compute the randomness value described above
+as the number of leading zeros among those 61 random bits.
+
+However, this would require modifying the W3C traceparent specification,
+therefore we do not propose to use bits of the TraceID.
+
+[This issue has been filed with the W3C trace context group.](https://github.com/w3c/trace-context/issues/463)
+
+### Not using TraceID hashing
+
+It would be possible to make a consistent sampling decision by hashing
+the TraceID, but we feel such an approach is not sufficient for making
+unbiased sampling decisions.  It is seen as a relatively difficult
+task to define and specify a good enough hashing function, much less
+to have it implemented in multiple languages.
+
+Hashing is also computationally expensive. This proposal uses extra
+data to avoid the computational cost of hashing TraceIDs.
+
+### Restriction to power-of-two
+
+Restricting head sampling rates to powers of two does not limit tail
+Samplers from using arbitrary probabilities.  The companion [OTEP
+170](https://github.com/open-telemetry/oteps/blob/main/text/trace/0170-sampling-probability.md) has discussed
+the use of a `sampler.adjusted_count` attribute that would not be
+limited to power-of-two values.  Discussion about how to represent the
+effective adjusted count for tail-sampled Spans belongs in [OTEP
+170](https://github.com/open-telemetry/oteps/blob/main/text/trace/0170-sampling-probability.md), not this OTEP.
+
+Restricting head sampling rates to powers of two does not limit
+Samplers from using arbitrary effective probabilities over a period of
+time.  For example, choosing 1/2 sampling half of the time and 1/4
+sampling half of the time leads to an effective sampling rate of 3/8.
```

Comment:
It may be better to pick a different example. p=3/8 looks neat, but what if I wanted p=7/16? I would need to pick p=1/2 vs. p=1/4 in proportion 3:1
```
1/2 * 0.75 + 1/4 * 0.25 = 7/16
```

Reply:
Perhaps an even better example would be to show that a more traditional 5% sampling rate can be obtained by sampling with 1/16 probability 60% of the time and with 1/32 probability 40% of the time.

Reply:
Applied.

Reply:
Done!